rust/http/kv-store/main.rs

```diff
@@ -0,0 +1,224 @@
+            .collect::<HashMap<String, String>>()
+    };
+    let params: HashMap<String, String> = req.uri().query().map_or(HashMap::new(), query_pairs);
+
+    // Validate action
+    let action = params.get("action").map(|s| s.as_str()).unwrap_or("get");
+    if !ALL_ACTIONS.contains(&action) {
+        return json_error(&format!(
+            "Invalid action '{}'. Supported actions: {}",
+            action,
+            ALL_ACTIONS.join(", ")
+        ));
+    }
+
+    // Validate store
+    let store_name = match params.get("store") {
+        Some(s) => s,
+        None => return json_error("Missing 'store' parameter"),
+    };
+
+    let store = match Store::open(store_name) {
+        Ok(s) => s,
+        Err(e) => return json_error(&format!("Failed to open store: {}", e)),
+    };
+
+    let response_json = match action {
+        "get" => {
+            let key = match params.get("key") {
+                Some(k) => k,
+                None => return json_error("Missing 'key' for get"),
+            };
+
+            match store.get(key) {
+                Ok(Some(v)) => {
+                    let value = String::from_utf8_lossy(&v);
+                    format!(
+                        r#"{{"Store":"{}","Action":"{}","Key":"{}","Response":"{}"}}"#,
+                        escape_json(store_name),
+                        escape_json(action),
+                        escape_json(key),
+                        escape_json(&value)
+                    )
+                }
+                Ok(None) => {
+                    format!(
+                        r#"{{"Store":"{}","Action":"{}","Key":"{}","Response":"not found"}}"#,
+                        escape_json(store_name),
+                        escape_json(action),
+                        escape_json(key)
+                    )
+                }
+                Err(e) => return json_error(&format!("error: {}", e)),
+            }
+        }
+        "scan" => {
+            let pattern = match params.get("match") {
+                Some(m) => m,
+                None => return json_error("Missing 'match' for scan"),
+            };
+
+            match store.scan(pattern) {
+                Ok(keys) => {
+                    let response = keys.join(", ");
+                    format!(
+                        r#"{{"Store":"{}","Action":"{}","Match":"{}","Response":"{}"}}"#,
+                        escape_json(store_name),
+                        escape_json(action),
+                        escape_json(pattern),
+                        escape_json(&response)
+                    )
+                }
+                Err(e) => return json_error(&format!("error: {}", e)),
+            }
+        }
+        "zrange" => {
+            let key = match params.get("key") {
+                Some(k) => k,
+                None => return json_error("Missing 'key' for zrange"),
+            };
+            let min = match params.get("min") {
```

Comment:
if min/max provided, but invalid, infinity is used by default, but when not provided, it is an error. I'd do it other way around - by default infinity, but if value is provided, it must be correct